packages/code-infra/src/utils/git.mjs

```diff
@@ -0,0 +1,68 @@
+import { $ } from 'execa';
+import gitUrlParse from 'git-url-parse';
+
+/**
+ * @typedef {Object} RepoInfo
+ * @property {string} owner - Repository owner
+ * @property {string} repo - Repository name
+ */
+
+/**
+ * Get current repository info from git remote
+ * @param {string[]} [remotes=['upstream', 'origin']] - Remote name(s) to check (default: ['upstream', 'origin'])
+ * @returns {Promise<RepoInfo>} Repository owner and name
+ */
+export async function getRepositoryInfo(remotes = ['upstream', 'origin']) {
+  /**
+   * @type {Record<string, string>}
+   */
+  const cause = {};
+  const cliResult = $`git remote -v`;
+  /**
+   * @type {Map<string, string>}
+   */
+  const repoRemotes = new Map();
+
+  for await (const line of cliResult) {
+    // Match pattern: "remoteName url (fetch|push)"
+    const match = line.trim().match(/^(?<remoteName>\S+)\s+(?<url>\S+)\s+\((?:fetch|push)\)$/);
```

Comment:
As `git remote get-url <name>` defaults to fetch, we should ignore all `(push)` lines. I also always recommend against regex if basic string manipulation can do the trick, it's too slow and unreadable.
```tsx
const [name, info] = line.trim().split('\t')
const [url, type] = info.split(' ')
if (type === '(fetch)') {
  repoRemotes.set(name, url)
} else if (type !== '(push)') {
  throw new Error(`Unexpected line format for "git remote -v": "${line}"`)
}
```

If necessary, we can later on add an option to emulate `git remote get-url --push <name>`.